app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx

```diff
@@ -1,47 +1,44 @@
-import { Component } from 'react';
+import arrayMutators from 'final-form-arrays';
+import { useState } from 'react';
+import { Field } from 'react-final-form';
+import { FieldArray } from 'react-final-form-arrays';
 import { Helmet } from 'react-helmet-async';
 import { Link } from 'react-router-dom';
-import { Field, FieldArray } from 'redux-form';
 import Button from 'app/components/Button';
 import { Content } from 'app/components/Content';
 import Dropdown from 'app/components/Dropdown';
-import {
-  TextInput,
-  SelectInput,
-  DatePicker,
-  legoForm,
-} from 'app/components/Form';
+import { TextInput, SelectInput, DatePicker } from 'app/components/Form';
+import LegoFinalForm from 'app/components/Form/LegoFinalForm';
 import Icon from 'app/components/Icon';
+import Flex from 'app/components/Layout/Flex';
 import { ConfirmModalWithParent } from 'app/components/Modal/ConfirmModal';
 import Time from 'app/components/Time';
 import type { EventType } from 'app/models';
 import type { SurveyEntity } from 'app/reducers/surveys';
 import { eventTypeToString, EVENT_CONSTANTS } from 'app/routes/events/utils';
+import {
+  DetailNavigation,
+  ListNavigation,
+  QuestionTypes,
+} from 'app/routes/surveys/utils';
+import { spySubmittable } from 'app/utils/formSpyUtils';
 import { createValidator, required } from 'app/utils/validation';
-import { DetailNavigation, ListNavigation, QuestionTypes } from '../../utils';
 import styles from '../surveys.css';
 import Question from './Question';
-import type { Element } from 'react';
-import type { FieldArrayProps, FormProps } from 'redux-form';
+import type { ReactNode } from 'react';
 
-type Props = FormProps & {
+type Props = {
   survey: SurveyEntity;
   autoFocus: Record<string, any>;
   surveyData: Array<Record<string, any>>;
-  submitFunction: (
-    arg0: SurveyEntity,
-    arg1: number | null | undefined
-  ) => Promise<any>;
+  submitFunction: (arg0: any) => Promise<any>;
   push: (arg0: string) => void;
   template?: Record<string, any>;
   selectedTemplateType?: EventType;
   destroy: () => void;
   initialize: () => void;
   activeFrom: string;
-};
-type State = {
-  templatePickerOpen: boolean;
-  templateTypeSelected: string;
+  initialValues: Record<string, any>;
```

Comment:
:cry:

Reply:
Do you know how big that object is? I'm not typing that